### ISSUE TITLE
Generating Mammography Reports from Multi-view Mammograms with BERT

### ISSUE THREAD

**Issue by reonaledo** [iRAIL-SNU/irail-paper-review#7]
### Motivation

- Mammography 영상의 판독문 작성 모델 개발과 관련된 첫 번째 연구.
- EfficientNet와 BERT 기반 Image to text 모델
- 검증은 generation metric과 doctor evaluation을 통해 적절히 수행했음
- 데이터: 러시아 병원 25K 개 검사
- 적당히 작동하는것 보이고 있지만 개선의 여지 많아보임
- 코드 공개. pretrained weight는 없음 [링크](https://github.com/sberbank-ai-lab/mammo2text/blob/master/vis_attention.ipynb)

### Method

![image](https://user-images.githubusercontent.com/58892327/218912579-c5ccd989-6669-4c19-9d1c-73b55a6382a6.png)
![image](https://user-images.githubusercontent.com/58892327/218912893-99e3df53-b73e-4d41-8a24-bb0025c9a880.png)

전체 구조
- 검사 당 4개의 영상은 파라메터를 공유하는 하나의 EfficientNet으로 각각 인코딩됨
- 디코더인 BERT는 Masked attention을 적용하여 decoding을 수행할 수 있게 함. 또한 사이에 이미지 feature와의 cross attention layer를 추가하였음. 

학습
22k개의 training set 이용
- Encoder pretraining
  - 판독문에서 7개의 binary 타겟 레이블(Lesions, Shadows, Calcifications, Fibrosis, Skin Thickening, BI-RADS>1, BI-RADS>2)을 간단한 정규표현식으로 도출
  - ImageNet에 사전훈련된 EfficientNET 기반 Image encoder를 위 레이블들에 대해 분류할 수 있도록 학습. Multilabel classification

- Decoder
  - GPT보다 BERT가 encoder-decoder 구조에 적합했다고 함. 
  - Pretrain: 러시아언어로 학습한 것과 medical corpus로 학습한(자세한정보없음) 것을 각자 사용하여 성능을 비교하고 있음.
  - 새로 추가한 Cross attention layer는 scratch 상태에서 report generation object로 학습되는듯
  - Cross attn layer에서 특정 텍스트 토큰에 대한 image region의 attention map을 시각화 할 수 있게 하였음. 

### Experiment & Result

![image](https://user-images.githubusercontent.com/58892327/218915158-b99b96e0-4e90-427a-b48a-8f7298673497.png)

모델 설명
- FEN2RND: 
  - ImageNet으로 사전훈련된 EfficientNet 사용. Random initialized BERT 사용. 4개의 view에 대해 인코더가 하나의 임베딩만 생성해서 디코더에서 어텐션하게 한 듯. 4개 view 아웃풋값을 평균때렸나? 1*768 임베딩만 생성하도록 하였음
- FEN2RND+att: 위 모델에서 4개 view 각각 임베딩을 생성하여 어텐션하게 함. 이하 2개의 모델도 마찬가지.  4988*768 생성
- MFEN2RUBERT: EfficientNet을 mammogram multilabel claissification으로 추가학습. BERT는 러시안에 사전훈련된 모델 사용 
- MFEN2MBERT: BERT를 medical corpus에 훈련한 것을 사용.

150개 test set에 대한 성능
- 기존 연구가 없어 random 모델과의 비교 및 ablation study만 진행
- Doctor evaluation
 - 한 mammography 당 총 6개의 판독문(Ground truth, random, 위4개 모델 )을 임의 순서로 1명의 의사에게 전달 후 평가함
 - CAL, LES: 생성 판독문의 Clacification과 Lesions에 대해 영상과 비교하여 생성판독문의 정확도 평가
  - Rating: 생성 판독문의 completness, relevance, accuracy에 대해 1~10점으로 점수 책정 후 normalize. 10점이 ground truth, 1점이 random 모델 oputput.
 

![image](https://user-images.githubusercontent.com/58892327/218918314-3e73bfde-cc12-43ff-b8dd-1dd9c61b7531.png)

Encoder pretrain 시 사용했던 multilabel classification 테스크의 성능이 BERT와의 vl train으로 개선되었는지 확인하는 실험.
Lesions, Shadow, Thick skin은 성능이 향상되었지만 Calcifications, Fibrosis는 오히려 조금 떨어짐. BERT와의 훈련에서 spacial information을 더 잘 이용하게 돼서 skin thickening같은 것들은 더 잘 예측할 수 있게 되었다고 설명. 

![image](https://user-images.githubusercontent.com/58892327/218918804-17ddf38d-4cc7-46b2-80bf-c3021f47d67e.png)

시각화 예시.
생성 판독문의 텍스트토큰에 대한 4개 view 이미지의 attention map. 
실제로 워킹하는지 내가 판단할 수 없지만 잘 된다면 모델 설명력에 유용할듯.

### Insight

전체적으로 MMG report generaiton 연구 시작을 위해 참고하기 좋은 논문이었음.
4개 view 각각의 reginal information을 생성해서 decoder에서 이를 이용함으로써 decoding 시 위치에 대한 정보를 직접적으로 이용하게 하고 추가적으로 attention map을 시각화할수있게 한 점이 인상깊었음.

개선하고싶은 점
- 데이터: 훈련 데이터가 25k. 너무 적다. 러시안 모델이다. 데이터 수 10만단위로 늘리고 영어데이터로 가자.ㅣ
- 모델 전체구조: 너무 기초적인 수준. 최신 연구 중 Coca의 구조가 적절해보임
![image](https://user-images.githubusercontent.com/58892327/218920612-a5a8b7e7-ce52-4403-8378-33c3f482caa1.png)
- image encoder: ViT 쓰고, multilabel classification과 더불어 판독문과의 CLIP loss도 함께 이용해서 사전학습.
- Text decoder: OPT같은 pretrained decoder 가져다 쓰기.
- 학습: coca처럼 contrasitve loss랑 captioning loss 사용. image encder의 아웃풋을 이용한 multilabel classification loss도 pretraining시에 유지시킬지는 고민.

CXR report generation 연구들과의 차이점:
- 대부분의 report generation연구는 CXR에 대해 이루어졌기 때문에 기술적으로 가장 발전되어있음.
- CXR연구들에서는 단순한 encoder-decoder 구조만을 사용하는것 보다 이미지에 대한 Higher concept을 디코딩단계에서 명시적으로 사용하는 Memory 기반 모듈을 같이 사용하는 것이 좋은 성능을 보이고 있음.  이는 판독문의 길이가 길고 CXR이 다양한 질병과 상태정보를 포함할 수 있기 때문임.
- MMG는 상대적으로 판독문이 정형화되어있고 포함하는 상태정보의 양이 적기 때문에 Memory 모듈이 CXR만큼은 필요 없을 수도 있음. 
- 하지만 MMG 이미지가 포함하는 정보들(Lesions, shadows, clacifications, fibrosis, skin thickneing, bi-rads, density 등)이 비교적 명시적으로 존재하기 때문에 memory 모듈의 적용이 충분히 시도해볼 가치가 있을 것으로 보임.

### Paper URL

https://aclanthology.org/2021.findings-emnlp.15/
